cmake/bob_robotics.cmake

```diff
@@ -365,17 +365,24 @@ macro(BoB_build)
         set(CMAKE_EXE_LINKER_FLAGS "-Wl,--allow-multiple-definition")
     endif()
 
-    # Use C++14. On Ubuntu 16.04, seemingly setting CMAKE_CXX_STANDARD doesn't
+    # If C++ standard has not been specified explicitly either in a
+    # CMakeLists.txt file or on the command line, set the standard to C++14, the
+    # minimum supported by BoB robotics.
```

Comment:
Unless we actually plan on updating BoB robotics to use a bunch of C++17 features I think it might be better to default to C++14 (as the comment says 😄) and set the standard to C++17 for gazebo projects